subsys/mgmt/mcumgr/smp/src/smp.c

```diff
@@ -461,7 +472,9 @@ int smp_process_request_packet(struct smp_streamer *streamer, void *vreq)
 	}
 
 	smp_free_buf(req, streamer->smpt);
-	smp_free_buf(rsp, streamer->smpt);
+	if (rsp) {
```

Comment:
Current version of `smp_free_buf` checks if rsp is NULL so the previous implementation seems ok to me. Is there a specific reason to change it now?

Reply:
Ah OK if it safe to call then check is not necessary. Client state never allocated response buffer. I will put that part back of original.